packages/contracts/deploy/007-OVM_L1CrossDomainMessenger.deploy.ts

```diff
@@ -2,55 +2,38 @@
 import { DeployFunction } from 'hardhat-deploy/dist/types'
 
 /* Imports: Internal */
-import { getDeployedContract } from '../src/hardhat-deploy-ethers'
+import {
+  getDeployedContract,
+  deployAndRegister,
+  waitUntilTrue,
+} from '../src/hardhat-deploy-ethers'
 
 const deployFn: DeployFunction = async (hre) => {
-  const { deploy } = hre.deployments
-  const { deployer } = await hre.getNamedAccounts()
-
   const Lib_AddressManager = await getDeployedContract(
     hre,
-    'Lib_AddressManager',
-    {
-      signerOrProvider: deployer,
-    }
+    'Lib_AddressManager'
   )
 
-  const result = await deploy('L1CrossDomainMessenger', {
-    from: deployer,
+  await deployAndRegister({
+    hre,
+    name: 'L1CrossDomainMessenger',
     args: [],
-    log: true,
+    postDeployAction: async (contract) => {
+      // Theoretically it's not necessary to initialize this contract since it sits behind
+      // a proxy. However, it's best practice to initialize it anyway just in case there's
+      // some unknown security hole. It also prevents another user from appearing like an
+      // official address because it managed to call the initialization function.
+      console.log(`Initializing L1CrossDomainMessenger...`)
+      await contract.initialize(Lib_AddressManager.address)
+
+      console.log(`Checking that contract was correctly initialized...`)
+      await waitUntilTrue(async () => {
+        return (
+          (await contract.libAddressManager()) === Lib_AddressManager.address
```

Comment:
Capitalization problem here potentially. I think its a good idea to make a helper function in `core-utils` that is like `compareAddresses(left | string, right | string): bool` that uses ethers to make sure both left and right are addresses and then compares them being checksummed